waku/v2/protocol/waku_store/waku_store.nim

```diff
@@ -391,8 +391,14 @@ proc query*(w: WakuStore, query: HistoryQuery, handler: QueryHandlerFunc) {.asyn
   #  - latency?
   #  - default store peer?
 
-  let peer = w.peers[0]
-  let connOpt = await w.peerManager.dialPeer(peer.peerInfo, WakuStoreCodec)
+  let peerOpt = w.peerManager.selectPeer(WakuStoreCodec)
+
+  if peerOpt.isNone():
+    error "failed to connect to remote peer"
```

Comment:
Or maybe there is no peer to connect to?

Reply:
Yes, thanks. That will be clearer.